metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/database/SraObjectsImporterThroughDatabase.java

```diff
@@ -116,45 +115,46 @@ public SraObjectsImporterThroughDatabase(
     }
 
     @Override
-    public Study importStudy(String accession) {
+    public Study importStudy(String accession) throws Exception {
         setEnaObjectQuery(EnaObjectQuery.STUDY_QUERY);
         Study study = super.importStudy(accession);
         setEnaObjectQuery(EnaObjectQuery.ANALYSIS_QUERY);
         return study;
     }
 
     @Override
-    public Analysis importAnalysis(String accession) {
+    public Analysis importAnalysis(String accession) throws Exception {
         setEnaObjectQuery(EnaObjectQuery.ANALYSIS_QUERY);
         return super.importAnalysis(accession);
     }
 
     @Override
-    public List<Sample> importSamples(AnalysisType analysisType) {
+    public List<Sample> importSamples(AnalysisType analysisType) throws Exception {
         setEnaObjectQuery(EnaObjectQuery.SAMPLE_QUERY);
         String analysisAccession = analysisType.getAccession();
         List<Sample> samples = new ArrayList<>();
-        try {
-            Map<String, String> idXmlMap = getSampleXmls(analysisAccession);
-            SampleType sampleType;
-            for (Map.Entry<String, String> entry : idXmlMap.entrySet()) {
+        Map<String, String> idXmlMap = getSampleXmls(analysisAccession);
+        SampleType sampleType;
+        for (Map.Entry<String, String> entry : idXmlMap.entrySet()) {
+            try {
                 sampleType = sraSampleXmlParser.parseXml(entry.getValue(), entry.getKey());
```

Comment:
Can you have a wrong xml and see entire operation is aborted as a test case

Reply:
its already tested that wrong XML throws XmlException in uk.ac.ebi.ampt2d.metadata.importer.xml  test package and also the current test cases are written like for any exceptions the import tree is aborted.

Reply:
If it verifies for which accession entire tree is aborted and nothing about root accession is imported its fine.